### ISSUE TITLE
New tag policy (?)

### ISSUE THREAD

**Issue by leamas** [Rasbats/shipdriver_pi#276]
When updating other plugins with the templates here things becomes much easier if it's possible to import all shipdriver tags to the plugin. This is no problem as long as the tag names don't collide.

With current scheme, shipdriver is using for example v1.2.3, and there is actually a rather high risk that will collide with a plugin tag with the same name. For that reason I propose that we instead start tagging shipdriver with tag names like sd1.2.3 instead. That is, to use a 'sd' prefix instead of 'v'.

While on it, the pre-release tags does not really work. This is just about workflow, but still. In the repo we have: 
```
v2.8.0.0-beta
v2.8.1.0-beta
v2.8.2.0-beta
v2.8.3.0-beta
v2.8.4.0-beta
v2.8.4.1-beta
```
This is just not how things are supposed to work. Any person viewing the tags, perhaps at github, will be confused. The more common and expected  flow would have been:
```
v2.8.5-beta1
v2.8.5-beta2
v2.8.5-beta3
v2.8.5
```
Or, what I actually prefer (but that is just personal preference):
```
v2.8.5-rc1
v2.8.5-rc2
v2.8.5-rc3
v2.8.5
```
The fourth digit should IMHO not really be used in the shipdriver tags. It is there for downstream packaging. That is, if another developer (why?) would take the shipdriver sources and make a new package. It is in this context just an obscure  corner case.

To summarize, I suggest that we stick to tags according to scheme above when we make a new 3.0 release. That is, something like
```
sd3.0-rc1
sd3.0-rc2
sd3.0-rc3
...
sd3.0          <= The actual 3.0 release
sd3.0.1        <= First update after 3.0
```

**Comment by Rasbats** [comment]
Now testing sd3.0.0-beta1. Closing.

**Comment by leamas** [comment]
If I could, I would reopen this...

Anyway, we are back on the wrong track. The repo now has

sd3.0.2-beta1
sd3.0.0-beta1

This makes no sense at all. The sd3.0.2-beta1 should have been sd3.0.0-beta2 to adhere to well-established conventions.  Is there something I miss here?

**Comment by leamas** [comment]
The more I look into this there are strange, complicated habits around. Why are you (and not only you...) changing the release for each and every build?

This is *very* strange and is not that common. As for the tarballs, they have names including the hash (to be unique) and the build number (to be sortable). 

If I would write instructions for _Making a release_ they would be something like this:

  1. Update version to that of the upcoming release (here, 3.0.0)
  2. Build a first beta release and test it.
  3. When the code seems stable enough: tag it with the beta1 tag (here, sd3.0.0-beta1) and push the tag
  4. Rebuild last commit on all builders. This will upload that build to the stable/ cloudsmith directories, using the tag's name.
  5. Rinse and repeat for beta2, beta3, etc.
  6. When the final release (here, 3.0) is ready to be released: build and test. And then do as with the betas, but this time with the sd3.0.0 tag.
  7. After releasing 3.0.0 (but not before): Update the version to 3.0.1 and continue development making untagged scratch builds.


Thoughts?

**Comment by Rasbats** [comment]
I am not, never has been, a programmer. I can only ape the habits of people I see working on these plugins. copy-paste developer would be a better description of my hobby.
Anyway ... I will delete the tags for sd3 and start again. Probably need to revert the commits as well.

**Comment by leamas** [comment]
OK, I think the problem here is that we actually have some bad examples  around. I suggest that you
  - revert the version in Plugin.cmake to 3.0.0 (and let it stay there for a while)
  - makes a build from that, with some simple testing that it is sound (nothing fancy...)
  - Removes the tags  sd3.*  (this is normally a thing never done, but should not be any harm here.
 
At this point you should have a build which sort of works, and nothing added after that. Tag and push:
   - git tag sd3.0.0-beta1
   - git push origin sd3.0.0-beta1
The last point is to rebuild the same commit in circleci and appveyour in the web interface. At that point the build the same thing, but the tarballs will got to the beta cloudsmith directory. This way you can keep control over what's actually uploaded to stable repos instead of that is just happens

Just my 5c, there are other ways for sure. That said, when we find a working flow we should document this somehow.

**Comment by leamas** [comment]
OK, all looks sound now. Good that you just dropped beta1 and went to beta 2
